lib/Support/BackedgeBuilder.cpp

```diff
@@ -25,6 +25,10 @@ void Backedge::setValue(mlir::Value newValue) {
   assert(!set && "backedge already set to a value!");
   value.replaceAllUsesWith(newValue);
   set = true;
+
+  // If the backedge is referenced again, it should now point to the updated
+  // value.
+  value = newValue;
```

Comment:
Hmmm I think this makes sense, but why didn't we need this before? This comes up when you use `backedge.value` after the edge has been resolved?

Reply:
> This comes up when you use backedge.value after the edge has been resolved?

Correct.

> Hmmm I think this makes sense, but why didn't we need this before?

My take is that it just hasn't been needed yet! I've actually been working with this "fix" locally for a while now, but have never gotten it checked in due to various PRs being dropped along the way.

Reply:
Ok great, well feel free to commit this part at any point I guess.